postpic/datahandling.py

```diff
@@ -1478,6 +1478,50 @@ def integrate(self, axes=None, method=scipy.integrate.simps):
         else:
             return self._integrate_scipy(axes, method)
 
+    def _derivative(self, axis):
+        from pkg_resources import parse_version
+        if parse_version(np.__version__) < parse_version('1.9'):
+            g = np.gradient(self, self.axes[axis].spacing)
+            if self.dimensions > 1:
+                g = g[axis]
+            der_field = self.replace_data(g)
+        else:
+            der_field = self.replace_data(np.gradient(self, self.axes[axis].spacing, axis=axis))
+
+        der_field.name = "{}'".format(self.name)
+        return der_field
+
+    def _derivative_stagger(self, axis):
+        oldax = self.axes[axis]
+        assert oldax.islinear()
```

Comment:
can you rather raise a proper Exception instead of using an assertion?

Reply:
Done. I also added another version check to allow derivative of non-linear axes starting with numpy 1.13.